packages/core/src/shared/tools.ts

```diff
@@ -735,3 +736,29 @@ export function generateRandomId(n: number = 21, alphabet?: string): string {
 
     return nanoid(n);
 }
+
+interface IStyleDataObject {
+    [key: string]: unknown;
+}
+
+/**
+ * compose styles by priority, the latter will overwrite the former
+ * @param { Nullable<IStyleData>[]} styles the styles to be composed
+ * @returns  { Nullable<IStyleData>[]}
+ */
+export function composeStyles(...styles: Nullable<IStyleData>[]): IStyleData {
```

Comment:
It is recommended to add unit tests for these util functions.

Reply:
it will be added before this PR be merged